### PR TITLE
Dance pole improvements, resolves #2368

### LINKED ISSUE #2368: Stripper poles are not part of the build system
#### Brief description of the bug
Dance poles are not part of the build system, despite being simple metal objects that make sense to be constructable.

#### What you expected to happen
Dance poles be constructable

#### What actually happened
Dance poles can only be moved, and not deconstructed or built.

#### Steps to reproduce
Look at the build menu for steel sheets, or attempt to deconstruct a stripper pole.

#### Code Revision
BYOND Version: 513.1527
Master commit: 233a2e3aa6176ba60f8c1818ba8a0b30acaa46ab 
Local commit: 25d7dba5a43218ce668063e98a04dacad0c1a81d
Server tools version: 4.5.3
 
#### Anything else you may wish to add:
I'm gonna implement it myself, the issue is so I can reference it later and not forget it's an ongoing issue.

### PR DESCRIPTION
**Pull request by RobotAlice**
## About The Pull Request

Implements deconstructing and constructing dance poles, also adds a get_description_interaction() based instructions blurb after the desc.

## Why It's Good For The Game

Not being able to destroy and build /obj/structures with steel is just wrong and unamerican.

## Changelog
:cl:
add: Steel sheet recipe to make dance poles (1 sheet)
add: Added construction/deconstruction functionality to dance poles
tweak: Fancy bluetext instructions for moving/removing dance poles
/:cl:

### CODE REVIEW

**Comment by lolman360** [comment]
people are going to spam them if they're only 1 steel

**Comment by RobotAlice** [comment]
Ah, shit. I'm going to have to move the recipe def into map stuff. I'll handle it when I wake up tomorrow.

File: maps/nsv_triumph/triumph_things.dm [hunk 1]
```diff
@@ -371,6 +371,22 @@ var/global/list/latejoin_shuttle   = list()
...
+			playsound(src, WT.usesound, 25, 1)
+			for (var/mob/M in viewers(src))
+				M.show_message("<span class='notice'>[user.name] deconstructed \the [src].</span>", 3, "<span class='notice'>You hear welding.</span>", 2)
+			new /obj/item/stack/material/steel(loc)
+			qdel(src)
+
+/obj/structure/dancepole/get_description_interaction()
+	var/list/results = list()
+	results += "[desc_panel_image("welder")] to deconstruct."
+	if(anchored) results += "[desc_panel_image("wrench")] to unbolt from the floor."
```
Comment (lolman360) [inline maps/nsv_triumph/triumph_things.dm:387]:
```suggestion
	if(anchored) 
		results += "[desc_panel_image("wrench")] to unbolt from the floor."
```

File: maps/nsv_triumph/triumph_things.dm [hunk 2]
```diff
@@ -371,6 +371,22 @@ var/global/list/latejoin_shuttle   = list()
...
+	else results += "[desc_panel_image("wrench")] to anchor to the floor."
```
Comment (lolman360) [inline maps/nsv_triumph/triumph_things.dm:388]:
```suggestion
	else 
		results += "[desc_panel_image("wrench")] to anchor to the floor."
```

**Review by lolman360 — CHANGES_REQUESTED** [review]

**Comment by lolman360** [comment]
also @RobotAlice just move the `structure/dancepole` define into its own dm or in the generic structures dm, minimizing snowflake code is good

**Comment by RobotAlice** [comment]
Yeah, I'll just throw the dancepole into the structures dm. I like inline if/else for simple clauses but I guess it's more idiomatic in dm like that. My rationale that 1 sheet is fine is because racks and table frames are also a sheet and nobody spams them, but if anyone else says it really needs to be higher I'll change that.

**Comment by Zandario** [comment]
👍

**Comment by lolman360** [comment]
it's more "heehoo funny stripper pole", table frames aren't inherently suggestive

inline if/else is cleaner sometimes but it's not in line with main's styleguide (which ostensibly applies here since more standardization good)

**Pull request merged**